addons/mail/static/src/models/composer/composer.js

```diff
@@ -289,82 +284,120 @@ function factory(dependencies) {
             body = this._generateMentionsLinks(body);
             body = parseAndTransform(body, addLink);
             body = this._generateEmojisOnHtml(body);
+            return body;
+        }
+
+        /**
+         * Recursive function that process the queue.
+         */
+        async processMessageQueue() {
+            if (!this.isProcessingQueue) {
+                this.update({
+                    isProcessingQueue: true,
+                });
+                const messages = this.thread.queue.filter((message) => !message.pendingError);
+                const message = messages[0];
+                try {
+                    if (message) {
+                        await this._postNextMessageInQueued(message);
```

Comment:
I don't like this arbitrary split between `processMessageQueue` and `_postNextMessageInQueued` because it makes it very strange to decide what is supposed to be handled in one or the other.

Reply:
It split the update of internal value and the posting logic. It also avoid having a 2 screens long function.
I would also have split the "crafting postData" and the RPC's in 2 functions but didn't since you already don't like it and it's not master.
I have remove the _postNext function.